pkg/oc/graph/genericgraph/graph.go

```diff
@@ -365,21 +342,21 @@ func (g Graph) AddEdge(from, to graph.Node, edgeKind string) {
 		kinds.Insert(g.EdgeKinds(existingEdge).List()...)
 	}
 
-	g.internal.SetEdge(NewEdge(from, to, kinds.List()...), 1.0)
+	g.internal.SetEdge(NewEdge(from, to, 1.0, kinds.List()...))
 }
 
 // addEdges adds the specified edges, filtered by the provided edge connection
 // function.
 func (g Graph) addEdges(edges []graph.Edge, fn EdgeFunc) {
 	for _, e := range edges {
 		switch t := e.(type) {
-		case concrete.WeightedEdge:
-			if fn(g, t.From(), t.To(), t.Edge.(Edge).Kinds()) {
-				g.internal.SetEdge(t.Edge.(Edge), t.Cost)
-			}
 		case Edge:
 			if fn(g, t.From(), t.To(), t.Kinds()) {
-				g.internal.SetEdge(t, 1.0)
+				g.internal.SetEdge(t)
+			}
+		case simple.Edge:
```

Comment:
It seems that before changes this function was able to handle only genericgraph.Edge. Is this case ever executed?